doc/man3/SSL_get0_peer_rpk.pod

```diff
@@ -0,0 +1,108 @@
+=pod
+
+=head1 NAME
+
+SSL_add_expected_rpk,
+SSL_get_negotiated_client_cert_type,
+SSL_get_negotiated_server_cert_type,
+SSL_get0_peer_rpk,
+SSL_SESSION_get0_peer_rpk - raw public key (RFC7250) support
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ int SSL_add_expected_rpk(SSL *s, EVP_PKEY *rpk, int nid);
+ int SSL_get_negotiated_client_cert_type(const SSL *s);
+ int SSL_get_negotiated_server_cert_type(const SSL *s);
+ EVP_PKEY *SSL_get0_peer_rpk(const SSL *s);
+ EVP_PKEY *SSL_SESSION_get0_peer_rpk(const SSL_SESSION *ss);
+
+=head1 DESCRIPTION
+
+SSL_add_expected_rpk() adds public key B<rpk> into the SSL B<s>'s DANE TLSA database.
+The B<nid> argument specifies the hash method to use.
```

Comment:
The DANE definitions were in `ssl_local.h`, and I didn't want to move them (yet). I don't like NIDs either (especially since OpenSSL is trying to move away from them - my understanding, and `NID_undef` doesn't make much sense). That being said, other than the final enumeration values, I suspect that the API is pretty close.

Reply:
The DANE API allows users to add new digest code points if in the future additional code points get registered. This is why I'd like to avoid having the list of valid code points hard-coded here. Either have the user pass an `EVP_MD` handle, or an arbitrary DANE `mtype` code point they know to be acceptable to the DANE layer.